core/src/main/java/org/mule/api/context/notification/FlowCallStack.java

```diff
@@ -20,12 +20,19 @@
 public interface FlowCallStack extends Serializable
 {
 
-    void push(FlowStackElement flowStackElement);
-
-    FlowStackElement pop();
-
+    /**
+     * @return the top-most element of this stack
+     */
     FlowStackElement peek();
 
-    Collection<FlowStackElement> getElementsCopy();
+    /**
+     * @return the elements of this stack as a list, ordered from top to bottom.
+     */
+    List<FlowStackElement> getElements();
+
+    /**
+     * @return the paths of the processors that were executed as part of this flow.
+     */
+    List<String> getExecutedProcessors();
```

Comment:
Is this needed? Looks like can be obtained from the FlowStackElements returned on #getElements